grassroots-backend/src/auth/Auth.controller.ts

```diff
@@ -9,63 +10,90 @@ import {
 } from "@nestjs/common";
 import { Response as ExpressResponse } from "express";
 import type { GrassrootsRequest } from "../types/GrassrootsRequest";
-import { ConfigService } from "@nestjs/config";
 import { LoginStateDTO } from "../grassroots-shared/LoginState.dto";
-import { VoidDTO } from "../grassroots-shared/Void.dto";
-import { ApiProperty, ApiQuery, ApiResponse } from "@nestjs/swagger";
+import { VoidDTO } from "../grassroots-shared/Void.dto"; // Add this import
+import { ApiTags, ApiQuery, ApiResponse } from "@nestjs/swagger";
 import { PublicRoute } from "./PublicRoute.decorator";
 import { OAuthGuard } from "./OAuth.guard";
+import { getEnvironmentVariables } from "../GetEnvironmentVariables";
 
 @Controller("auth")
+@ApiTags("auth")
 export class AuthController {
-  constructor(private configService: ConfigService) {}
+  private environmentVariables: Record<string, string> | undefined;
+
+  constructor() {
+    void this.loadEnvironmentVariables();
+  }
+
+  private async loadEnvironmentVariables(): Promise<void> {
+    this.environmentVariables = await getEnvironmentVariables();
+  }
 
   // The frontend can redirect here to trigger login.
   @Get("login")
   @UseGuards(OAuthGuard)
   @PublicRoute()
   @ApiQuery({ name: "redirect_path", type: String })
-  login(@Query() redirect_path: string): VoidDTO {
+  @ApiResponse({ status: 200, type: VoidDTO })
+  async login(@Query("redirect_path") redirectPath: string): Promise<VoidDTO> {
```

Comment:
It looks like you're fixing a bunch of eslint violations that I can't trigger locally.

Not sure what's going on here.